GVFS/GVFS.UnitTests/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,137 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using GVFS.Platform.Mac;
+using GVFS.Service;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Mock.FileSystem;
+using Moq;
+using NUnit.Framework;
+using System.IO;
+
+namespace GVFS.UnitTests.Service
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class MacServiceTests
+    {
+        private const string GVFSServiceName = "GVFS.Service";
+        private const string ExpectedActiveUserId = "502";
+        private static readonly string ExpectedActiveRepoPath = Path.Combine("mock:", "code", "repo2");
+        private static readonly string ServiceDataLocation = Path.Combine("mock:", "registryDataFolder");
+
+        private MockFileSystem fileSystem;
+        private MockTracer tracer;
+        private Mock<MacPlatform> gvfsPlatformMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.tracer = new MockTracer();
+            this.fileSystem = new MockFileSystem(new MockDirectory(ServiceDataLocation, null, null));
+            this.gvfsPlatformMock = new Mock<MacPlatform>();
```

Comment:
Can we use `MockBehavior.Strict` and call `VerifyAll` in the `TearDown`?  If not why?

Reply:
There are 3 test cases in `MacServiceTests.cs`. 
- `ServiceStartTriggersAutoMountForCurrentUser` - Uses `IRepoRegistry` mock to verify that `Service` startup calls `AutoMountRepos`. Updated `IRepoRegistry` Mock to use `MockBehavior.Strict` & `VerifyAll`

- `RepoRegistryMountsOnlyRegisteredRepos` - Uses `IRepoMounter` mock to verify that `RepoRegistry` calls `MountRepository` with the right args. Updated `IRepoMounter` Mock to use `MockBehavior.Strict` & `VerifyAll`

- `MountProcessLaunchedUsingCorrectArgs` - Uses `MountLauncher` mock to verify that `IRepoMounter` calls `LaunchProcess` and `WaitUntiMounted` with expected args. Updated `MountLauncher` Mock to use `MockBehavior.Strict` & `VerifyAll`

None of these test cases exercise all methods that are setup in `GVFSPlatform` Mock inside `SetUp()`. So using `MockBehavior.Strict` in `GVFSPlatform` Mock and `VerifyAll` in `TearDown()` will cause test failures.

Reply:
Also, `GVFSPlatform` is a generic class with so many methods and interfaces. From the testing perspective what is important is `AutoMountRepos`, `MountRepository`, `LaunchProcess` and `WaitUntilMounted` methods get called by the `SUT` with the right arguments. The `GVFSPlatform` methods that `SUT`s internally use is private to the `SUT` and not a subject of the tests. Thoughts?